src/hooks.ts

```diff
@@ -11,21 +11,38 @@ import {useMemo} from "react";
 export function useQueryWithClient<TQueryFnData = unknown>(
     queryKey: string | readonly unknown[],
     queryFn: (client: IDeskproClient) => Promise<TQueryFnData>,
-    options?: Omit<UseQueryOptions<TQueryFnData, unknown, TQueryFnData, string | readonly unknown[]>, 'queryKey' | 'queryFn'>
-): UseQueryResult<TQueryFnData> {
+    options?: Omit<
+      UseQueryOptions<
+        TQueryFnData,
+        unknown,
+        TQueryFnData,
+        string | readonly unknown[]
+      >,
+      "queryKey" | "queryFn"
+    >
+  ): UseQueryResult<TQueryFnData> {
     const { client } = useDeskproAppClient();
-
+  
     const key = Array.isArray(queryKey) ? queryKey : [queryKey];
-
+  
     return useQuery(
-        key,
-        () => (client && queryFn(client)) as Promise<TQueryFnData>,
-        {
-            ...(options ?? {}),
-            enabled: options?.enabled === undefined ? !! client : (client && options?.enabled),
-        } as Omit<UseQueryOptions<TQueryFnData, unknown, TQueryFnData, string | readonly unknown[]>, 'queryKey' | 'queryFn'>
+      key,
+      () => (client && queryFn(client)) as Promise<TQueryFnData>,
+      {
+        ...(options ?? {}),
+        enabled: options?.enabled ? !!client : !!client && options?.enabled,
+        suspense: false,
```

Comment:
@iTweeno please confirm that setting `suspense: false` does not affect suspense; both loading state and error boundary.

I would not expect suspense to affect query caching -- but if this is the case, feels like there's another undiscovered problem

Reply:
It unfortunately is the latter, the requests just retrigger with suspense on, am looking into it

Reply:
thanks David

Reply:
my hunch is it may be something to do with passing `client` to `queryFn`, do you know if react query is caching based on query arguments as well as the key?

Reply:
I believe its only the key but ill check.
But lets say it does, why does suspense change anything?

Reply:
no idea.

may be something to do with `client` becoming available, causing a re-render and somehow interacting with suspense?

I guess things to check are (with suspense enabled and then disabled):

- client availability causing re-render
- query keys somehow changing on re-render
- `enabled` flag being toggled too many times when the `client` is available